src/main/java/uk/gov/hmcts/reform/jobscheduler/services/jobs/ActionSerializer.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.hmcts.reform.jobscheduler.services.jobs;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.jobscheduler.model.HttpAction;
+
+@Component
+public class ActionSerializer {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    // Quartz can store only basic types as job data, we serialize it to json string.
+    public String serialize(HttpAction action) {
+        try {
+            return objectMapper.writeValueAsString(action);
+        } catch (JsonProcessingException exc) {
+            throw new JobActionSerializationException("Unable to serialize action", exc);
+        }
+    }
+
+    public static class JobActionSerializationException extends RuntimeException {
+        public JobActionSerializationException(String message, Throwable cause) {
```

Comment:
considering previous comment - can be package private

Reply:
I also like it better when it's in a separate file. Not too important, though.